src/oncall/scheduler/default.py

```diff
@@ -486,6 +426,12 @@ def populate(self, schedule, start_time, dbinfo):
         handoff = start_epoch + timedelta(seconds=first_event_start)
         handoff = timezone(schedule['timezone']).localize(handoff)
 
+        # create temporary table if function is being used to preview
```

Comment:
We should limit the scope of the populate function. `populate()` shouldn't be responsible for determining whether the action is a preview or a populate request, and it shouldn't include falcon's `req` or `resp` as arguments. Instead, populate should do exactly that, populate the schedule. 

We can go back to creating a `preview` function, which can handle this logic. Both `preview()` and `populate()` can rely on another helper `schedule_events()` (or something similar), which can generate the populate event data. Then, `preview` can pass this info back to the caller in `api/v0/preview.py` and `populate` can update the DB. The falcon responder in `api/v0/preview.py` can then be responsible for formatting and updating the HTTP response with the schedule preview data.